modules/hdf4_handler/hdfutil.cc

```diff
@@ -107,7 +107,7 @@ void *ExportDataForDODS(const hdf_genvec & v, int i)
 {
     switch (v.number_type()) {
     case DFNT_INT16:{
-            int16 *temp = new int16;
+            auto temp = new int16;
             *temp = v.elt_int16(i);
             return (void *) temp;
```

Comment:
This is another 'curious' question: If you used `return temp;` instead of the cast, would C++-11 make the cast for you given the return type of the function is `void*`? Again, more of a question than anything else.